src/lib/hash-importer.ts

```diff
@@ -25,10 +31,19 @@ export default async function getHasherForCode (code: number): Promise<Supported
       return from({
         name: 'keccak-256',
         code,
-        encode: async (data: Uint8Array) => {
+        encode: async (data: Uint8Array): Promise<Uint8Array> => {
           return new Uint8Array(keccak256.arrayBuffer(data))
         }
       })
+    case 30: // blake3-multihash
+      return from({
+        name: 'blake3-multihash',
+        code,
+        encode: async (data: Uint8Array): Promise<Uint8Array> => {
+          const { digest } = await blake3Digest(data)
```

Comment:
You could pull the import down into this so we only import this library dynamically when we need it, and add as a peer/optional/optionalPeer. That may solve the ipfs-webui issue.